grand_challenge_dicom_de_identifier/deidentifier.py

```diff
@@ -225,61 +248,105 @@ def _handle_element(
             action = action_desc["default"]
             justification = action_desc.get("justification", "")
 
-        if action == ActionKind.REMOVE:
-            del dataset[elem.tag]
-        elif action == ActionKind.KEEP:
-            pass
-        elif action == ActionKind.REJECT:
-            raise RejectedDICOMFileError(justification=justification) from None
-        elif action == ActionKind.UID:
-            elem.value = self.uid_map[elem.value]
-        elif action in (ActionKind.REPLACE, ActionKind.REPLACE_0):
-            if elem.VR == "SQ":  # Sequence
-                for sequence_item in elem.value:
-                    sequence_item.walk(
-                        partial(
-                            self._handle_element,
-                            action_lookup={},  # Always defer to the default
-                            default_action={
-                                "default": action,
-                                "justification": "Parent sequence was replaced",
-                            },
-                        )
+        action_map: Dict[str, Callable[[ActionContext], None]] = {
+            ActionKind.REMOVE: self._handle_remove_action,
+            ActionKind.KEEP: self._handle_keep_action,
+            ActionKind.REJECT: self._handle_reject_action,
+            ActionKind.UID: self._handle_uid_action,
+            ActionKind.REPLACE: self._handle_replace_action,
+            ActionKind.REPLACE_0: self._handle_replace_0,
+        }
+        try:
+            handler = action_map[action]
+        except KeyError:
+            raise NotImplementedError(
+                f"Action {action} not implemented"
+            ) from None
+
+        handler(
+            ActionContext(
+                dataset=dataset,
+                elem=elem,
+                action=action,
+                justification=justification,
+            )
+        )
+
+    def _handle_remove_action(self, /, context: ActionContext) -> None:
```

Comment:
`self` is always a positional argument so the slash is a bit useless here, right?